subsys/nrf_security/src/drivers/cracen/cracenpsa/src/platform_keys/platform_keys.c

```diff
@@ -0,0 +1,285 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include "cracen_psa.h"
+#include "cracen_psa_primitives.h"
+#include "platform_keys.h"
+#include <stdint.h>
+#include <psa/crypto.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/sys/util.h>
+
+/**
+ * PSA Key ID scheme
+ *
+ * N0  Generation
+ * N1  Reserved
+ * N2  Usage p0
+ * N3  Usage p1
+ * N4  Domain p0
+ * N5  Domain p1
+ * N6  Access rights
+ * N7  0x4
+ */
+
+#define USAGE_FWENC	0x20
+#define USAGE_PUBKEY	0x21
+#define USAGE_AUTHDEBUG 0x23
+#define USAGE_STMTRACE	0x25
+#define USAGE_COREDUMP	0x25
+
+#define DOMAIN_NONE	   0x00
+#define DOMAIN_SECURE	   0x01
+#define DOMAIN_APPLICATION 0x02
+#define DOMAIN_RADIO	   0x03
+#define DOMAIN_CELL	   0x04
+#define DOMAIN_ISIM	   0x05
+#define DOMAIN_WIFI	   0x06
+#define DOMAIN_SYSCTRL	   0x08
+
+typedef struct sicr_key {
+	uint32_t nonce[3];
+	uint8_t *nonce_addr;
+	psa_key_type_t type;
+	psa_key_bits_t bits;
+	uint8_t *attr_addr;
+	uint8_t *key_buffer;
+	size_t key_buffer_max_length;
+	uint8_t *mac;
+	size_t mac_size;
+} sicr_key;
+
+static sicr_key find_key(uint32_t id)
+{
+	struct sicr_key key = {};
+
+	uint32_t usage = (id >> 8) & 0xff;
```

Comment:
Since we have a representation of the id which is from MSB to LSB:
8 bits: Not used
8 bits: Domain
8 bits: Usage
4 bits: Not used
4 bits: Generation 

Can we make the representation into a packed struct or something so that we dont' have to do the shift + &  every time?

If that's not possible maybe we can add some defines for the shifting of the different fields?

Reply:
no. I'll make defines